### DISCUSSION TITLE
Asking a Doubt!

### DISCUSSION THREAD

**Discussion by felixoder** [Cyfrin/foundry-full-course-cu#2670] [Q&A] [answered]
Let me tell about the doubt directly so as we know receive and fallback when people somehow tries to send me money without calling the handeling function lets say createFund() 

I got that perfectly! now my question is If I call createFund() directly and send money the gas required to do so is "less" than the gas if I use receive and fallback somehow! 

Can someone please clear that out that why this takes more gas!

## Just adding the boiler plate code to make everyone understand

``` function createFund() public payable {
      // The stuffs
    }
receive() external payable { 
        createFund();
    }

    fallback() external payable { 
        createFund();
    }

```

If I try to send money without calling the createFund() I get more gas instead I create transaction using the createFund() directly!

**Comment by EngrPips** [comment, marked as answer]
Hello @felixoder, The reason is when you call the `createFund` function directly, the contract executes the `createFund` process, but when you don't directly call the `createFund` function, then the contract needs to first process either the `receive` or `fallback` function after that it will now start processing the `createFund` function. Generally for the `EVM` with more processes executed more gas is consumed. Please ask questions if anything is not clear.

**Reply by felixoder** [reply]
Nice point! More processes more gas! 
Thanks bro @EngrPips

**Reply by EngrPips** [reply]
You are welcome, Friend. Keep crushing it.
